### PR TITLE
Add swagger config and API docs

### PR DESCRIPTION
**Pull request by vickyhella**
## Summary

- Adds the openapi plugin `2.0.0-beta.3`
- Bumps react to 18.2.0 to avoid plugin incompatibilities
- Adds swagger config to the `docusaurus.config.js` file
- Generates API mdx files for v1.2 and v1.1 (run `yarn docusaurus clean-api-docs all && yarn docusaurus gen-api-docs all && yarn docusaurus clean-api-docs:version api:all && yarn docusaurus gen-api-docs:version api:all`)
- Uses autogenerated sidebars for both docs and API pages

## Related issues and PRs

This PR addresses https://github.com/harvester/harvester/issues/3042 and #373, and replaces #242

## Screenshots

> Clicking the version switch on the API pages will switch to the API pages of the specified version (v1.1 and v1.2 only).

image

## References

- Plugin documentation: [docusaurus-openapi.tryingpan.dev](https://docusaurus-openapi.tryingpan.dev/)
- Official demo: [PaloAltoNetworks/docusaurus-openapi-docs@main/demo](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/tree/main/demo?rgh-link-date=2022-11-09T09%3A31%3A28Z)
- Example json/yaml config: [PaloAltoNetworks/docusaurus-openapi-docs@main/demo/examples](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/tree/main/demo/examples?rgh-link-date=2022-11-09T09%3A31%3A28Z)
- Example docusaurus.config.js config: [PaloAltoNetworks/docusaurus-openapi-docs@main/demo/docusaurus.config.js](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/main/demo/docusaurus.config.js?rgh-link-date=2022-11-09T09%3A31%3A28Z)

### CODE REVIEW

**Comment by chrisho** [comment]
On the right side tab of each API page, is the URL expected? Should It contain the path?
image

File: scripts/ignore-openapi [hunk 1]
```diff
@@ -11,18 +13,20 @@ AFTER_COMMIT_ID=${DRONE_COMMIT}
...
 
 # if the api/swaggers.json has changed
-if echo "$FILES" | grep -q "api/swagger.json"; then
+if echo "$FILES" | grep -qE "v(.*)swagger.json"; then
     # perform re-generate and check for dirty files
-    yarn clean-api-docs all
-    yarn gen-api-docs all
-    dirty_files=$(git status --porcelain --untracked-files=no)
+    yarn clean-api-docs all && yarn gen-api-docs all
+    yarn docusaurus clean-api-docs:version api:all && yarn docusaurus gen-api-docs:version api:all
```
Comment (guangbochen) [inline scripts/ignore-openapi:19]: is this redundant? does version also includes the default one?

Comment (chrisho) [reply scripts/ignore-openapi]: Removed redundant command line.

**Review by guangbochen — APPROVED** [review]
LGTM, please squash some commits where applicable before the PR merge, thanks.

**Review by bk201 — APPROVED** [review]
lgtm, thanks!

**Review by LucasSaintarbor — APPROVED** [review]

**Pull request merged**